src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/CSharpCodeRefactoringVerifier`1+Test.cs

```diff
@@ -85,8 +92,70 @@ public Test()
             public string? EditorConfig { get; set; }
 
 #if !CODE_STYLE
+            private readonly List<AdhocWorkspace> _workspaces = new();
+
             protected override AnalyzerOptions GetAnalyzerOptions(Project project)
                 => new WorkspaceAnalyzerOptions(base.GetAnalyzerOptions(project), project.Solution);
+
+            /// <summary>
+            /// The <see cref="TestHost"/> we want this test to run in.  Defaults to <see cref="TestHost.InProcess"/> if unspecified.
+            /// </summary>
+            public TestHost TestHost { get; set; } = TestHost.InProcess;
+
+            /// <summary>
+            /// The set of code action <see cref="CodeAction.Title"/>s offered the user in this exact order.
+            /// Set this to ensure that a very specific set of actions is offered (or not).
+            /// </summary>
+            public string[]? ExactActionSetOffered { get; set; }
+
+            protected override ImmutableArray<CodeAction> FilterCodeActions(ImmutableArray<CodeAction> actions)
```

Comment:
This method is under `#if !CODE_STYLE`. Was that intended?

Reply:
Yes.  for now, i'm only modifying the testing abilities outside of the CODE_STYLE layer.  @sharwell  to advise if we'd want this to be for both layers.  I'm deferring to him on all design decisions for this test harness.

Reply:
This one should work in both. `CreateWorkspace` is restricted because the OOP compositions are not available in code style layer testing.

Reply:
sgtm.